pinot-core/src/main/java/com/linkedin/pinot/core/operator/docvalsets/UnSortedSingleValueSet.java

```diff
@@ -44,6 +46,19 @@ public DataType getValueType() {
   }
 
   @Override
+  public void getStringValues(int[] inDocIds, int inStartPos, int inDocIdsSize, String[] outValues, int outStartPos) {
+    ReaderContext context = sVReader.createContext();
+    int inEndPos = inStartPos + inDocIdsSize;
+    try {
+      for (int i = inStartPos; i < inEndPos; i++) {
+        outValues[outStartPos++] = sVReader.getString(inDocIds[i], context);
```

Comment:
For the thread safe issue, if possible, do the duplicate() here will get higher performance and produce lesser garbage.